<launch>

    <!-- the camera -->

    <include file="$(find azure_kinect_ros_driver)/launch/kinect_rgbd.launch">
        <arg name="depth_enabled" value="true" />
        <arg name="depth_mode" value="WFOV_UNBINNED" />
        <arg name="fps" value="15" />
        <arg name="color_resolution" value="1536P" />
        <arg name="num_worker_threads" value="4" />
        <arg name="overwrite_robot_description" value="false" />
    </include>

    <!-- the Apriltap detection -->

    <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="launch_prefix" default="" />

    <!-- configure camera input -->
    <arg name="camera_name" default="/rgb" />
    <arg name="image_topic" default="image_raw" />
    <arg name="queue_size" default="1" />

    <!-- apriltag_ros continuous detection node -->
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node"
        name="apriltag_ros_continuous_node" clear_params="true" output="screen"
        launch-prefix="$(arg launch_prefix)">
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default:
        false -->
        <param name="queue_size" type="int" value="$(arg queue_size)" />

        <!-- load parameters (incl. tag family, tags, etc.) -->
        <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" />
        <rosparam command="load" file="$(find leap_task_A)/config/tags_handeye_calib.yaml" />
    </node>

    <!-- Leap Hand -->

    <include file="$(find leap_hardware)/launch/leap_bringup.launch">
        <arg name="free_move" value="true" />
        <arg name="enable_visualization" value="true" />
    </include>

    <node pkg="leap_hardware" name="leaphand_publisher" type="leap_joint_state_publisher.py"
        output="screen" />

    <!-- easy_eye_hand -->

    <include file="$(find easy_handeye)/launch/calibrate.launch">
        <arg name="eye_on_hand" value="false" />

        <!-- free move and disable moveit -->
        <arg name="freehand_robot_movement" value="true" />

        <!-- you can choose any identifier, as long as you use the same for publishing the
        calibration -->
        <arg name="namespace_prefix" value="leap_kinect_calib" />

        <!-- fill in the following parameters according to your robot's published tf frames -->
        <arg name="robot_base_frame" value="palm_lower" />
        <arg name="robot_effector_frame" value="fingertip_3" />

        <!-- fill in the following parameters according to your tracking system's published tf
        frames -->
        <arg name="tracking_base_frame" value="camera_base" />
        <arg name="tracking_marker_frame" value="AprilTag" />
    </include>


</launch>